src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java

```diff
@@ -120,11 +123,13 @@ public TransactionDao(Jdbi jdbi) {
         this.jdbi = jdbi;
     }
 
-    public Optional<TransactionEntity> findTransactionByExternalIdAndGatewayAccountId(String externalId, String gatewayAccountId) {
+    public Optional<TransactionEntity> findTransactionByExternalIdAndGatewayAccountId(String externalId, String gatewayAccountId, TransactionType transactionType, String parentTransactionExternalId) {
```

Comment:
Method name could change as it's purpose for transactions with more filters

Reply:
yeah, wasn't sure how to name it.

Reply:
May be 
```suggestion
    public Optional<TransactionEntity> findTransaction(String externalId, String gatewayAccountId, TransactionType transactionType, String parentTransactionExternalId) {
```